internal/pkg/cli/pipeline_status.go

```diff
@@ -4,72 +4,211 @@
 package cli
 
 import (
+	"errors"
 	"fmt"
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/codepipeline"
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/session"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/cli/selector"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/config"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy/cloudformation/stack"
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/describe"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/manifest"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+	"io"
 
 	"github.com/spf13/cobra"
 )
 
+const (
+	pipelineStatusAppNamePrompt          = "Which application's pipeline's status would you like to show?"
+	pipelineStatusAppNameHelpPrompt      = "An application is a collection of related services."
+	fmtPipelineStatusPipelineNamePrompt  = "Which pipeline of %s would you like to show the status of?"
+	pipelineStatusPipelineNameHelpPrompt = "The details of a pipeline's status will be shown (e.g., stages, status, transition)."
+)
+
 type pipelineStatusVars struct {
 	*GlobalOpts
 	shouldOutputJSON bool
-	appName          string
 	pipelineName     string
 }
 
 type pipelineStatusOpts struct {
 	pipelineStatusVars
 
-	pipelineSvc         pipelineStateGetter
-	statusDescriber     pipelineStatusDescriber
-	initStatusDescriber func(opts *pipelineStatusOpts) error
+	w             io.Writer
+	ws            wsPipelineReader
+	store         store
+	pipelineSvc   pipelineStateGetter
```

Comment:
As mentioned above, it looks like you should be able to just use the existing `pipelineGetter` interface here, since you don't appear to be calling `GetPipelineState` within this package.